h2o-algos/src/main/java/hex/schemas/IsolationForestV3.java

```diff
@@ -42,5 +43,8 @@
         @API(help = "Number of variables randomly sampled as candidates at each split. If set to -1, defaults (number of predictors)/3.", gridable = true)
         public int mtries;
 
+        @API(help = "Contamination ratio - the proportion of anomalies in the input dataset. If undefined (-1) the predict function will not mark observations as anomalies and only anomaly score will be returned. Defaults to -1 (undefined).")
+        public double contamination;
```

Comment:
It'd be nice if the parameter got checked in `init` method:

Here, I offer a patch to quickly apply:

```patch
Index: h2o-algos/src/main/java/hex/tree/isofor/IsolationForest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- h2o-algos/src/main/java/hex/tree/isofor/IsolationForest.java	(revision e0bdc1ed2cd95be4389be2cfb5ddc01f1ec005ea)
+++ h2o-algos/src/main/java/hex/tree/isofor/IsolationForest.java	(date 1592320031449)
@@ -70,6 +70,10 @@
       throw new IllegalStateException("Isolation Forest doesn't expect the distribution to be specified by the user");
     }
     _parms._distribution = DistributionFamily.gaussian;
+
+    if(_parms._contamination < -1){
+      throw new IllegalArgumentException(String.format("Illegal contamination value: '%d'. Minimum possible value is '-1' - undefined.", _parms._contamination));
+    }
   }
 
   @Override
```

Reply:
good idea, thank you!